extension/demo/src/modules/storage.js

```diff
@@ -0,0 +1,23 @@
+'use strict'
+
+export async function set (data) {
+  console.log(data)
+  await chrome.storage.local.set(data, console.log)
+
+  return !chrome.runtime.lastError
+}
```

Comment:
Thanks to the [polyfill](https://github.com/mozilla/webextension-polyfill#using-the-promise-based-apis) we don't need to check for `lastError`. Instead, simply return the promise and if we need to handle rejections/failures we can simply nest the call to this function (`storage.set`) with a try/catch.

Reply:
That seems less practical than returning a bool

Reply:
It isn't. The above just magics away any error messages which we might want to do some times but definitely not always.

Most places that consume this function would simply do `const results = await storage.set(...)` without any fallback. If we want to make sure that we recover from any potential errors in production we should probably do that globally or at least via some sort helper/decoration. As of now there's not really any reason to though.

The only place where we need/want the error to be handled right now is in the response to setting the access token. It's better to handle it there.